torchrl/objectives/a2c.py

```diff
@@ -333,14 +334,19 @@ def loss_critic(self, tensordict: TensorDictBase) -> torch.Tensor:
             )
         return self.critic_coef * loss_value
 
+    @property
+    @cache_values
+    def __detach_critic_params(self):
+        return self.critic_params.detach()
+
```

Comment:
Here self.critic_params is already using cache so we are only caching the detach operation right?

Reply:
yep

Reply:
detaching can take quite some time bc we need to rebuild a hugely nested tensordict
caching is virtually memory free and alleviates that